ci/buildkite-pipeline.sh

```diff
@@ -14,12 +14,34 @@ if [[ -n $CI_PULL_REQUEST ]]; then
   pr_number=${BASH_REMATCH[1]}
   echo "get affected files from PR: $pr_number"
 
-  # Fetch the number of commits in the PR
-  commits_no=$(gh pr view "$pr_number" --json commits --jq '.commits | length')
-  echo "number of commits in this PR: $commits_no"
+  # shellcheck disable=SC2016
+  # shellcheck disable=SC2046
+  # ref: https://github.com/cli/cli/issues/5368#issuecomment-1087515074
+  query='
+  query($owner: String!, $repo: String!, $pr: Int!, $endCursor: String) {
+    repository(owner: $owner, name: $repo) {
+      pullRequest(number: $pr) {
+        files(first: 100, after: $endCursor) {
+          pageInfo{ hasNextPage, endCursor }
+          nodes {
+            path
+          }
+        }
+      }
+    }
+  }'
 
   # get affected files
-  readarray -t affected_files < <(git diff HEAD~"$commits_no"..HEAD --name-status | cut -f2)
+  readarray -t affected_files < <(
+    gh api graphql \
+      -f query="$query" \
+      -F pr="$pr_number" \
+      -F owner='anza-xyz' \
+      -F repo='agave' \
```

Comment:
afaik we can parse it from `BUILDKITE_REPO` but I think we should be fine with current status 🤔

Reply:
Is it non-trivial?
Not insisting, just wondering.
In general, the more constants we hard-code the more fragile things become.
But we obviously want to weight it against the parsing complexity.

And, in particular, we have other repos that use this code, that would fail to run this script correctly, if these values are hardcoded.

Reply:
https://github.com/anza-xyz/agave/pull/2075/commits/ff378dbd572f1db88a187d45b013b2d48e1fc9ee